### PR TITLE
add empty section for compile options

### LINKED ISSUE #1739: IAR compiler not supported in CMake
**Describe the bug**
Current the CMake assumes that the user is either using MSVC, clang or gnu compilers as defined in this file. If the compiler is MSVC or clang then it assumes it's GNU.
https://github.com/Azure/azure-sdk-for-c/blob/de29097e07a2a9d7268e2dc6cd0b2a6e5ea8c495/eng/cmake/global_compile_options.txt#L20

When trying to compile with IAR, the copmile fails as the IAR compiler doesnt recognise the GNU compiler flags.

***Exception or Stack Trace***
NA

**To Reproduce**
1. Install IAR compiler
1. Recursive clone the [GSG ](https://github.com/azure-rtos/getting-started)
1. Configure the IAR path [here](https://github.com/azure-rtos/getting-started/blob/04e4d630b8a13a162083e051e6fabd0e11299097/cmake/arm-iar-cortex-toolchain.cmake#L4)
1. Change to "STMicroelectronics/STM32L4_L4+"
1. Run `cmake -Bbuild -GNinja -DCMAKE_TOOLCHAIN_FILE="../../cmake/arm-iar-cortex-m4.cmake"`
1. Run `cmake --build build`
1. Observe compiler error due to GNU compiler flags

***Code Snippet***
There error is from here, where it assumes the compiler is GNU even when it isnt.
https://github.com/Azure/azure-sdk-for-c/blob/de29097e07a2a9d7268e2dc6cd0b2a6e5ea8c495/eng/cmake/global_compile_options.txt#L20

**Expected behavior**
Only add GNU compiler flags if using the GNU compiler. If the compiler is unknown to the C-SDK, then dont add additional compiler flags.

**Screenshots**
NA

**Setup (please complete the following information):**
- OS: Windows
- IDE: CMake
- master

**Additional context**
NA

**Information Checklist**
Please make sure that you have added all the following information above and checkoff the required fields otherwise we will treat the issuer as an incomplete report

- [x] Bug Description Added
- [x] Repro Steps Added
- [x] Setup information Added

### PR DESCRIPTION
**Pull request by danewalton**
Closes #1739 

Until we have gate support for IAR, defaulting to not adding flags is useful to have wider compatibility.
If/when we do, for IAR flag reference, here starting on page 123:
https://www.mvtool.co.kr/hb/data/pds/EWARM_CompilerReference.pdf

### CODE REVIEW

File: eng/cmake/global_compile_options.txt [hunk 1]
```diff
@@ -17,10 +17,12 @@ elseif(CMAKE_C_COMPILER_ID MATCHES "Clang")
   endif()
 
   add_compile_options(-Xclang -Wall -Wextra -pedantic  ${WARNINGS_AS_ERRORS_FLAG} -Wdocumentation -Wdocumentation-unknown-command -fcomment-block-commands=retval -Wcast-qual -Wunused -Wuninitialized -Wmissing-declarations -Wconversion -Wpointer-arith -Wshadow -Wfloat-equal)
-else()
+elseif(CMAKE_C_COMPILER_ID MATCHES "GCC")
```
Comment (ryanwinter) [inline eng/cmake/global_compile_options.txt:20]: replace GCC  with GNU

Comment (danewalton) [reply eng/cmake/global_compile_options.txt:20]: Ah you're right my apologies
https://cmake.org/cmake/help/latest/variable/CMAKE_LANG_COMPILER_ID.html

**Comment by ryanwinter** [comment]
LGTM!

File: eng/cmake/global_compile_options.txt [hunk 2]
```diff
@@ -17,10 +17,12 @@ elseif(CMAKE_C_COMPILER_ID MATCHES "Clang")
...
   add_compile_options(-Xclang -Wall -Wextra -pedantic  ${WARNINGS_AS_ERRORS_FLAG} -Wdocumentation -Wdocumentation-unknown-command -fcomment-block-commands=retval -Wcast-qual -Wunused -Wuninitialized -Wmissing-declarations -Wconversion -Wpointer-arith -Wshadow -Wfloat-equal)
-else()
+elseif(CMAKE_C_COMPILER_ID MATCHES "GNU")
   if(WARNINGS_AS_ERRORS)
     set(WARNINGS_AS_ERRORS_FLAG "-Werror")
   endif()
 
   add_compile_options(-Wall -Wextra -pedantic  ${WARNINGS_AS_ERRORS_FLAG} -Wcast-qual -Wunused -Wuninitialized -Wmissing-declarations -Wconversion -Wpointer-arith -Wshadow -Wlogical-op -Wfloat-equal)
+else()
+  message(WARNING "Using an unsupported compiler. Stricter compiler flags are not turned on.")
```
Comment (RickWinter) [inline eng/cmake/global_compile_options.txt:27]:
```suggestion
  message(WARNING "Using an unsupported compiler. Disabling stricter compiler flags.")
```

**Review by RickWinter — APPROVED** [review]

**Pull request merged**